python/ql/src/experimental/dataflow/internal/DataFlowPublic.qll

```diff
@@ -157,16 +170,18 @@ class ParameterNode extends EssaNode {
  *
  * It is important that all extending classes in scope are disjoint.
  */
-class BarrierGuard extends Expr {
-  // /** Holds if this guard validates `e` upon evaluating to `v`. */
-  // abstract predicate checks(Expr e, AbstractValue v);
+class BarrierGuard extends GuardNode {
+  /** Holds if this guard validates `e` upon evaluating to `v`. */
+  abstract predicate checks(ControlFlowNode node, boolean testIsTrue);
```

Comment:
Having a boolean called `testIsTrue` rubs me wrong, so I suggest we pick up the naming from other languages here. C++ uses `b`, Java uses `branch`.

```suggestion
  /** Holds if this guard validates `node` upon evaluating to `branch`. */
  abstract predicate checks(ControlFlowNode node, boolean branch);
```